src/librbd/migration/S3Stream.cc

```diff
@@ -89,8 +89,31 @@ void S3Stream<I>::open(Context* on_finish) {
   }
 
   m_url = url_value.get_str();
+
+  librados::Rados rados(m_image_ctx->md_ctx);
+  int r = 0;
   m_access_key = access_key.get_str();
+  if (util::is_config_key_uri(m_access_key)) {
+    r = util::get_config_key(rados, m_access_key, &m_access_key);
```

Comment:
We are going to block our work threads regardless since we don't have a librados async version. It's non-ideal but it only happens once at image open time.